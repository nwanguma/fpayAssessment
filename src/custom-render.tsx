import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

const Wrapper: React.FC = ({ children }) => {
  return <MemoryRouter>{children}</MemoryRouter>;
};

const customRender = (ui: any, options?: any) =>
  render(ui, { wrapper: Wrapper, ...options });

export * from "@testing-library/react";

export { customRender as render };
