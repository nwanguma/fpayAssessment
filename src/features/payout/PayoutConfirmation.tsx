import { State } from "./payout-reducer";

interface IProps {
  payoutDetails: State;
  handleSubmit: () => void;
}

interface Classes {
  section: string;
  sectionItem: string;
  sectionItemLabel: string;
  sectionItemDetail: string;
  sectionItemDetailBold: string;
  btn: string;
}

const classes: Classes = {
  section: "container py-3 border-solid border-b border-white-border",
  sectionItem: "container flex flex-col md:flex-row justify-between py-2 px-1",
  sectionItemLabel: "text-[13px] text-grey-label",
  sectionItemDetail: "text-right text-[13px] text-grey-medium",
  sectionItemDetailBold: "text-right text-[16px] font-medium text-grey-medium",
  btn: "btn w-[100%] text-white-main bg-green-button",
};

const Confirmation: React.FC<IProps> = ({ payoutDetails, handleSubmit }) => {
  return (
    <div className="card">
      <div className="container">
        <h3 className="card-heading__primary">
          Review details of your transfer
        </h3>
        <div className={classes.section}>
          <div className={classes.sectionItem}>
            <p className={classes.sectionItemLabel}>You send</p>
            <p className={classes.sectionItemDetailBold}>
              {payoutDetails.userAmount} {payoutDetails.baseCurrency}
            </p>
          </div>
          <div className={classes.sectionItem}>
            <p className={classes.sectionItemLabel}>Total fees (included)</p>
            <p className={classes.sectionItemDetail}>
              {payoutDetails.fee} {payoutDetails.baseCurrency}
            </p>
          </div>
          <div className={classes.sectionItem}>
            <p className={classes.sectionItemLabel}>Amount weâ€™ll convert</p>
            <p className={classes.sectionItemDetail}>
              {payoutDetails.amountMinusFee} {payoutDetails.baseCurrency}
            </p>
          </div>
          <div className={classes.sectionItem}>
            <p className={classes.sectionItemLabel}>Guaranteed rate</p>
            <p className={classes.sectionItemDetail}>{payoutDetails.rate}</p>
          </div>
          <div className={classes.sectionItem}>
            <p className={classes.sectionItemLabel}>
              {payoutDetails.fullname.split(" ")[0]} gets
            </p>
            <p className={classes.sectionItemDetailBold}>
              {payoutDetails.convertedAmount} {payoutDetails.targetCurrency}
            </p>
          </div>
        </div>
        <div className={classes.section}>
          <div className={classes.sectionItem}>
            <p className={classes.sectionItemLabel}>Name</p>
            <p className={classes.sectionItemDetail}>
              {payoutDetails.fullname}
            </p>
          </div>
          <div className={classes.sectionItem}>
            <p className={classes.sectionItemLabel}>Email address</p>
            <p className={classes.sectionItemDetail}>{payoutDetails.mail}</p>
          </div>
          <div className={classes.sectionItem}>
            <p className={classes.sectionItemLabel}>IBAN / Account number</p>
            <p className={classes.sectionItemDetail}>
              {payoutDetails.accountNumber}
            </p>
          </div>
        </div>
        <button className={classes.btn}>Confirm and continue</button>
      </div>
    </div>
  );
};

export default Confirmation;
