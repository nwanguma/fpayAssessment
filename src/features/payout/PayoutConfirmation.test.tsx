import { render, screen } from "../../custom-render";

import PayoutConfirmation from "./PayoutConfirmation";

describe("PayoutConfirmation", () => {
  const payoutDetails = {
    userAmount: "657",
    feeRate: 0.00369,
    fee: 6.65,
    amountMinusFee: 300,
    baseCurrency: "EUR",
    rates: {},
    rate: 30,
    targetCurrency: "NGN",
    convertedAmount: "509",
    fullname: "Will Tybur",
    mail: "willthemantybur@hotmail.com",
    accountNumber: "67679767676",
    swiftCode: "",
  };

  test("<PayoutConfirmation /> matches snapshot", () => {
    const component = render(
      <PayoutConfirmation payoutDetails={payoutDetails} />
    );

    expect(component.container).toMatchSnapshot();
  });

  test("It renders correctly", () => {
    render(<PayoutConfirmation payoutDetails={payoutDetails} />);

    expect(screen.getByTestId("card")).toBeInTheDocument();
  });

  test("It contains a button", () => {
    render(<PayoutConfirmation payoutDetails={payoutDetails} />);

    expect(
      screen.getByRole("button", { name: /Confirm and continue/i })
    ).toBeInTheDocument();
  });

  test("It contains the correct form data", () => {
    render(<PayoutConfirmation payoutDetails={payoutDetails} />);

    expect(
      screen.getByText(/willthemantybur@hotmail.com/i)
    ).toBeInTheDocument();
    expect(screen.getByText(/Will Tybur/i)).toBeInTheDocument();
    expect(screen.getByText(/NGN/i)).toBeInTheDocument();
    expect(screen.getByText(/67679767676/i)).toBeInTheDocument();
    expect(screen.queryByText(/chlamydia/i)).not.toBeInTheDocument();
  });
});

export {};
