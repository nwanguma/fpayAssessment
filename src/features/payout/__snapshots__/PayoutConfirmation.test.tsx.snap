// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayoutConfirmation <PayoutConfirmation /> matches snapshot 1`] = `
<div>
  <div
    class="card"
    data-testid="card"
  >
    <div
      class="container"
    >
      <h3
        class="card-heading__primary"
      >
        Review details of your transfer
      </h3>
      <div
        class="container py-3 border-solid border-b border-white-border"
      >
        <div
          class="container flex flex-col md:flex-row justify-between py-2 px-1"
        >
          <p
            class="text-[13px] text-grey-label"
          >
            You send
          </p>
          <p
            class="text-right text-[16px] font-medium text-grey-medium"
          >
            657
             
            EUR
          </p>
        </div>
        <div
          class="container flex flex-col md:flex-row justify-between py-2 px-1"
        >
          <p
            class="text-[13px] text-grey-label"
          >
            Total fees (included)
          </p>
          <p
            class="text-right text-[13px] text-grey-medium"
          >
            6.65
             
            EUR
          </p>
        </div>
        <div
          class="container flex flex-col md:flex-row justify-between py-2 px-1"
        >
          <p
            class="text-[13px] text-grey-label"
          >
            Amount weâ€™ll convert
          </p>
          <p
            class="text-right text-[13px] text-grey-medium"
          >
            300
             
            EUR
          </p>
        </div>
        <div
          class="container flex flex-col md:flex-row justify-between py-2 px-1"
        >
          <p
            class="text-[13px] text-grey-label"
          >
            Guaranteed rate
          </p>
          <p
            class="text-right text-[13px] text-grey-medium"
          >
            30
          </p>
        </div>
        <div
          class="container flex flex-col md:flex-row justify-between py-2 px-1"
        >
          <p
            class="text-[13px] text-grey-label"
          >
            Will
             gets
          </p>
          <p
            class="text-right text-[16px] font-medium text-grey-medium"
          >
            509
             
            NGN
          </p>
        </div>
      </div>
      <div
        class="container py-3 border-solid border-b border-white-border"
      >
        <div
          class="container flex flex-col md:flex-row justify-between py-2 px-1"
        >
          <p
            class="text-[13px] text-grey-label"
          >
            Name
          </p>
          <p
            class="text-right text-[13px] text-grey-medium"
          >
            Will Tybur
          </p>
        </div>
        <div
          class="container flex flex-col md:flex-row justify-between py-2 px-1"
        >
          <p
            class="text-[13px] text-grey-label"
          >
            Email address
          </p>
          <p
            class="text-right text-[13px] text-grey-medium"
          >
            willthemantybur@hotmail.com
          </p>
        </div>
        <div
          class="container flex flex-col md:flex-row justify-between py-2 px-1"
        >
          <p
            class="text-[13px] text-grey-label"
          >
            IBAN / Account number
          </p>
          <p
            class="text-right text-[13px] text-grey-medium"
          >
            67679767676
          </p>
        </div>
      </div>
      <button
        class="btn w-[100%] text-white-main bg-green-button"
      >
        Confirm and continue
      </button>
    </div>
  </div>
</div>
`;
